sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/CommentedCodeCheck.kt

```diff
@@ -54,12 +51,13 @@ class CommentedCodeCheck : AbstractCheck() {
             }
         })
 
-        val codeVerifier = KotlinCodeVerifier()
         groupedComments.forEach { comments ->
-            val content = comments.joinToString("\n") { it.getContent() }
-            if (codeVerifier.containsCode(content)) {
-                val textRanges = comments.map { kotlinFileContext.textRange(it) }
-                kotlinFileContext.reportIssue(kotlinFileContext.merge(textRanges), "Remove this commented out code.")
+            if (file.firstChild !in comments) {
```

Comment:
The problem is that we can have multiple AST nodes belonging to the header, so when the header comment uses multiple lines of single-line comments (`//`) instead of simply a multi-line comment (`/*...*/`). Unfortunately, this seems to be a thing in the wild. So before grouping them, it's not easy to determine whether a single-line comment that is not the first node in the file, is part of the header.